lib/rules/no-extra-boolean-cast.js

```diff
@@ -174,9 +217,18 @@ module.exports = {
                 case "DoWhileStatement":
                 case "WhileStatement":
                 case "ForStatement":
+                case "SequenceExpression":
                     return false;
                 case "ConditionalExpression":
-                    return precedence(node) <= precedence(parent);
+                    if (previousNode === parent.test) {
+                        return precedence(node) <= precedence(parent);
+                    }
+                    if (previousNode === parent.consequent || previousNode === parent.alternate) {
+                        return precedence(node) < precedence({ type: "AssignmentExpression" });
```

Comment:
Can you also add one test where this returns `true`?

Reply:
Good spot! - Added a test... The only thing that can be less is the comma operator, so I've added that.